Error compiling zcash with flag --disable-wallet

When trying to compile zcash v5.2.0 with the wallet disabled I get the following error:

```
Options used to compile and link:
  with wallet   = no
  with zmq      = yes
  with test     = yes
  use asm       = yes
  sanitizers    = 
  debug enabled = yes
  gprof enabled = no
  werror        = no

  target os     = linux
  build os      = 

  CC            = /workspace/debug-no-wallet/depends/x86_64-pc-linux-gnu/share/../native/bin/ccache /workspace/debug-no-wallet/depends/x86_64-pc-linux-gnu/native/bin/clang -target x86_64-pc-linux-gnu -B/workspace/debug-no-wallet/depends/x86_64-pc-linux-gnu/native/bin -m64
  CFLAGS        = -pipe -O0 
  CPPFLAGS      =  -DDEBUG -DDEBUG_LOCKORDER  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Qunused-arguments -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_PEDANTIC -I/workspace/debug-no-wallet/depends/x86_64-pc-linux-gnu/share/../include/  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS
  CXX           = /workspace/debug-no-wallet/depends/x86_64-pc-linux-gnu/share/../native/bin/ccache /workspace/debug-no-wallet/depends/x86_64-pc-linux-gnu/native/bin/clang++ -target x86_64-pc-linux-gnu -B/workspace/debug-no-wallet/depends/x86_64-pc-linux-gnu/native/bin -stdlib=libc++ -m64 -std=c++17
  CXXFLAGS      =  -O0 -g3 -ftrapv  -Wformat -Wformat-security -Wstack-protector -fstack-protector-all -Wthread-safety-analysis   -pipe -O0  -fno-strict-aliasing
  LDFLAGS       = -Qunused-arguments -pthread  -Wl,-z,relro -Wl,-z,now -pie  -fuse-ld=lld -L/workspace/debug-no-wallet/depends/x86_64-pc-linux-gnu/share/../lib  -static-libstdc++ -lc++abi
  ARFLAGS       = cr

+ make -j4
Making all in src
make[1]: Entering directory '/workspace/debug-no-wallet/src'
  GEN      rust/gen/include/rust/blake2b.h
  GEN      rust/gen/include/rust/bundlecache.h
  GEN      rust/gen/include/rust/equihash.h
  GEN      rust/gen/include/rust/orchard_bundle.h
  GEN      rust/gen/include/rust/sapling.h
  GEN      rust/gen/include/rust/wallet_scanner.h
make  all-recursive
make[2]: Entering directory '/workspace/debug-no-wallet/src'
make[3]: Entering directory '/workspace/debug-no-wallet/src'
  CXX      zcashd-bitcoind.o
  CXX      libbitcoin_server_a-sendalert.o
  CXX      libbitcoin_server_a-addrman.o
  CXX      libbitcoin_server_a-addrdb.o
  CXX      libbitcoin_server_a-alert.o
  CXX      libbitcoin_server_a-asyncrpcoperation.o
  CXX      libbitcoin_server_a-asyncrpcqueue.o
  CXX      libbitcoin_server_a-bloom.o
  CXX      libbitcoin_server_a-chain.o
  CXX      libbitcoin_server_a-checkpoints.o
  CXX      libbitcoin_server_a-deprecation.o
  CXX      libbitcoin_server_a-experimental_features.o
  CXX      libbitcoin_server_a-httprpc.o
  CXX      libbitcoin_server_a-httpserver.o
  CXX      libbitcoin_server_a-init.o
  CXX      libbitcoin_server_a-dbwrapper.o
  CXX      libbitcoin_server_a-main.o
  CXX      libbitcoin_server_a-merkleblock.o
  CXX      libbitcoin_server_a-metrics.o
  CXX      libbitcoin_server_a-miner.o
init.cpp:1527:17: error: use of undeclared identifier 'string'
    std::vector<string> uacomments;
                ^
1 error generated.
```
I tried to reproduce the error with 4 combinations using the script `zcutil/build.sh`

`$ CONFIGURE_FLAGS="--enable-wallet=no" ./zcutil/build.sh`
`$ CONFIGURE_FLAGS="--enable-wallet=no --enable-debug" ./zcutil/build.sh`

`$ CONFIGURE_FLAGS="--disable-wallet" ./zcutil/build.sh`
`$ CONFIGURE_FLAGS="--disable-wallet --enable-debug" ./zcutil/build.sh`

In all 4 combinations I get the same error.

Machine specs:
- OS: Debian 11 - Docker image https://hub.docker.com/r/electriccoinco/zcashd-worker-debian-11
- CPU: Intel x86_64 (GCP machine c2-standard-16)
- Linux kernel version: 5.10.133
- Compiler version: gcc (Debian 10.2.1-6) 10.2.1 20210110
- Linker version: GNU ld (GNU Binutils for Debian) 2.35.2
- Assembler version: GNU assembler (GNU Binutils for Debian) 2.35.2

Fixing the `string` error, gets us farther, but still fails:
```
  CXXLD    zcash-gtest
ld.lld: error: undefined symbol: bitdb
>>> referenced by utils.cpp:43 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:43)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())
>>> referenced by utils.cpp:67 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:67)
>>>               gtest/zcash_gtest-utils.o:(UnloadGlobalWallet())

ld.lld: error: undefined symbol: CDBEnv::Flush(bool)
>>> referenced by utils.cpp:43 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:43)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())
>>> referenced by utils.cpp:67 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:67)
>>>               gtest/zcash_gtest-utils.o:(UnloadGlobalWallet())

ld.lld: error: undefined symbol: CDBEnv::Reset()
>>> referenced by utils.cpp:44 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:44)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())
>>> referenced by utils.cpp:68 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:68)
>>>               gtest/zcash_gtest-utils.o:(UnloadGlobalWallet())

ld.lld: error: undefined symbol: CDBEnv::MakeMock()
>>> referenced by utils.cpp:46 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:46)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())

ld.lld: error: undefined symbol: pwalletMain
>>> referenced by utils.cpp:48 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:48)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())
>>> referenced by utils.cpp:63 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:63)
>>>               gtest/zcash_gtest-utils.o:(UnloadGlobalWallet())

ld.lld: error: undefined symbol: CWallet::LoadWallet(bool&)
>>> referenced by utils.cpp:49 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:49)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())

ld.lld: error: undefined symbol: CWallet::GenerateNewSeed(Language)
>>> referenced by utils.cpp:51 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:51)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())

ld.lld: error: undefined symbol: CWallet::VerifyMnemonicSeed(std::__1::basic_string<char, std::__1::char_traits<char>, secure_allocator<char> > const&)
>>> referenced by utils.cpp:52 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:52)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())

ld.lld: error: undefined symbol: RegisterWalletRPCCommands(CRPCTable&)
>>> referenced by utils.cpp:56 (/home/greg/Documents/ECC/zcash-rpc-binary-name/src/gtest/utils.cpp:56)
>>>               gtest/zcash_gtest-utils.o:(LoadGlobalWallet())

ld.lld: error: undefined symbol: vtable for CWallet
>>> referenced by wallet.h:1313 (./wallet/wallet.h:1313)
>>>               gtest/zcash_gtest-utils.o:(CWallet::CWallet(CChainParams const&))
>>> the vtable symbol may be undefined because the class is missing its key function (see https://lld.llvm.org/missingkeyfunction)

ld.lld: error: undefined symbol: vtable for WalletBatchScanner
>>> referenced by wallet.h:1064 (./wallet/wallet.h:1064)
>>>               gtest/zcash_gtest-utils.o:(CWallet::CWallet(CChainParams const&))
>>> the vtable symbol may be undefined because the class is missing its key function (see https://lld.llvm.org/missingkeyfunction)

ld.lld: error: undefined symbol: WalletBatchScanner::CreateBatchScanner(CWallet*)
>>> referenced by wallet.h:1064 (./wallet/wallet.h:1064)
>>>               gtest/zcash_gtest-utils.o:(CWallet::CWallet(CChainParams const&))

ld.lld: error: undefined symbol: vtable for CCryptoKeyStore
>>> referenced by crypter.h:160 (./wallet/crypter.h:160)
>>>               gtest/zcash_gtest-utils.o:(CCryptoKeyStore::CCryptoKeyStore())
>>> referenced by crypter.h:132 (./wallet/crypter.h:132)
>>>               gtest/zcash_gtest-utils.o:(CCryptoKeyStore::~CCryptoKeyStore())
>>> the vtable symbol may be undefined because the class is missing its key function (see https://lld.llvm.org/missingkeyfunction)
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [Makefile:3761: zcash-gtest] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/home/greg/Documents/ECC/zcash-rpc-binary-name/src'
make[2]: *** [Makefile:8797: all-recursive] Error 1
make[2]: Leaving directory '/home/greg/Documents/ECC/zcash-rpc-binary-name/src'
make[1]: *** [Makefile:2784: all] Error 2
make[1]: Leaving directory '/home/greg/Documents/ECC/zcash-rpc-binary-name/src'
make: *** [Makefile:645: all-recursive] Error 1
```

--- a/src/gtest/utils.cpp
+++ b/src/gtest/utils.cpp
@@ -1,6 +1,8 @@
 #include "gtest/utils.h"
 #include "rpc/server.h"
+#ifdef ENABLE_WALLET
 #include "wallet/wallet.h"
+#endif
 
 int GenZero(int n)
 {
@@ -35,6 +37,8 @@ void LoadProofParameters() {
     );
 }
 
+#ifdef ENABLE_WALLET
+
 void LoadGlobalWallet() {
     CCoinsViewDB *pcoinsdbview;
     bool fFirstRun;
@@ -67,3 +71,5 @@ void UnloadGlobalWallet() {
     bitdb.Flush(true);
     bitdb.Reset();
 }
+
+#endif

--- a/src/init.cpp
+++ b/src/init.cpp
@@ -1524,7 +1524,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     RegisterNodeSignals(GetNodeSignals());
 
     // sanitize comments per BIP-0014, format user agent and check total size
-    std::vector<string> uacomments;
+    std::vector<std::string> uacomments;
     for (std::string cmt : mapMultiArgs["-uacomment"])
     {
         if (cmt != SanitizeString(cmt, SAFE_CHARS_UA_COMMENT))

